n2y/converter.py

```diff
@@ -0,0 +1,466 @@
+
+
+class NumberedListItem(Block):
+    def __init__(self, client: Client, block, get_children=True):
+        super().__init__(client, block, get_children)
+        self.text = RichTextArray(self.text)
+
+    def to_pandoc(self):
+        content = [Plain(self.text.to_pandoc())]
+        children = super().to_pandoc()
+        if children:
+            content.extend(children)
+        return content
+
+
+class NumberedList(Block):
+    def __init__(self, client: Client, block, get_children=True):
+        self.client = client
+        self.has_children = False
+        self.items = []
+        self.created_time = None
+        self.last_edited_time = None
+        self.type = "numbered_list"
+        self.id = None
+
+    def append(self, item: NumberedListItem):
+        self.items.append(item)
+
+    def to_pandoc(self):
+        return OrderedList((1, Decimal(), Period()), [i.to_pandoc() for i in self.items])
+
+
+class HeadingBase(Block):
+    def __init__(self, client: Client, block, get_children=True):
+        super().__init__(client, block, get_children)
+        self.text = RichTextArray(self.text)
+
+    def to_pandoc(self):
+        return Header(self.level, ('', [], []), self.text.to_pandoc())
+
+
+class HeadingOne(HeadingBase):
+    level = 1
+
+
+class HeadingTwo(HeadingBase):
+    level = 2
+
+
+class HeadingThree(HeadingBase):
+    level = 3
+
+
+class Divider(Block):
+    def to_pandoc(self):
+        return HorizontalRule()
+
+
+class Bookmark(Block):
+    def to_pandoc(self):
+        caption = None
+        if self.caption:
+            caption = RichTextArray(self.caption).to_pandoc()
+        else:
+            caption = [Str(self.url)]
+        return Para([Link(('', [], []), caption, (self.url, ''))])
+
+
+class CodeBlockFenced(Block):
+    def to_pandoc(self):
+        return CodeBlock(('', [self.language], []), self.text[0]['plain_text'])
+
+
+class Quote(Block):
+    def to_pandoc(self):
+        return BlockQuote([Para(RichTextArray(self.text).to_pandoc())])
+
+
+class ImageBlock(Block):
+    def __init__(self, client: Client, block, get_children=True):
+        super().__init__(client, block, get_children)
+        self.file = File(block['image'])
+
+    def to_pandoc(self):
+        url = None
+        if self.file.type == "external":
+            url = self.file.url
+        elif self.file.type == "file":
+            url = self.file.download()
+        caption = RichTextArray(self.caption)
+        return Para([Image(('', [], []), caption.to_pandoc(), (url, ''))])
+
+
+class File():
+    def __init__(self, obj):
+        if obj['type'] == "file":
+            self.type = "file"
+            self.url = obj['file']['url']
+            self.expiry_time = obj['file']['expiry_time']
+        elif obj['type'] == "external":
+            self.type = "external"
+            self.url = obj['external']['url']
+
+    def download(self):
+        # TODO: append created time as hex to end of file to prevent collisions?
```

Comment:
Sorry, I was misreading my own code. For some reason, I thought this was naming the page and not file attachments. md5 should be fine for files since they changed less frequently.